packages/integration-sdk-runtime/src/logger/index.ts

```diff
@@ -158,6 +161,24 @@ export class IntegrationLogger extends EventEmitter
     this._errorSet = input.errorSet;
   }
 
+  /**
+   * Answers `true` when the err has been reported to the logger instance
+   * through these functions:
+   *
+   * * warn(err, ...)
+   * * error(err, ...)
+   *
+   * This can be used by outer `try/catch` blocks to avoid logging the same
+   * error twice:
+   *
+   * ```js
+   * if (!logger.isHandledError(err)) {
+   *   logger.error(err);
+   * }
```

Comment:
I don't care for this pattern. Why can't the logger just consult whether it's seen an err before and avoid re-logging it? This is closely related to the question, why does the SDK re-throw errors that it handles?